src/client/history/history.js

```diff
@@ -116,22 +117,31 @@ spf.history.add = function(opt_url, opt_state, opt_doCallback) {
  *     entry.  When the user returns to this entry, the "state" property of the
  *     event will contain a copy of this object.
  * @param {boolean=} opt_doCallback Whether to do the history event callback.
- * @param {boolean=} opt_retainState Whether to retain existing state if no new
- *     state is provided. Defaults to false.
  * @throws {Error} If the state object is too large. For example, Firefox will
  *     pass the object to JSON.stringify and impose a 640k character limit.
  * @throws {Error} If the URL is not in the same domain, a SECURITY_ERR
  *     (code == 18) is thrown.
  * @throws {Error} If window.history.replaceState is not a function.
  */
-spf.history.replace = function(opt_url, opt_state, opt_doCallback,
-                               opt_retainState) {
+spf.history.replace = function(opt_url, opt_state, opt_doCallback) {
+  var state = null;
+  // Set the existing state values.
   var currentState = spf.history.getCurrentState_();
-  if (opt_retainState && currentState) {
-    opt_state = opt_state || currentState;
+  if (currentState) {
+    state = state || {};
```

Comment:
At this point state should alway be null. It is probably safe to just do state = {};